zipline-loader/src/commonMain/kotlin/app/cash/zipline/loader/internal/cache/ZiplineCache.kt

```diff
@@ -367,4 +381,19 @@ internal class ZiplineCache internal constructor(
   private fun path(metadata: Files): Path {
     return directory / "entry-${metadata.id}.bin"
   }
+
+  /**
+   * Update file record freshAt timestamp to reflect that the manifest is still seen as fresh
+   */
+  fun updateManifestFreshAt(applicationName: String, loadedManifest: LoadedManifest) {
+    val freshAtMs = loadedManifest.freshAtEpochMs ?: throw IllegalArgumentException(
```

Comment:
Could this be a `!!` ? It’s all internal callers

Maybe a better way to describe this... can we trigger this being null in calling code?
 - if we can, use a nice exception as you have here, and write a test to confirm it
 - if we can’t, use `!!` and save some lines of code

As-is, it reads like this is a thing that can actually happen and I don’t think it can

Reply:
Good point, I missed updating that. Ended up that this case isn't possible so no exception or `!!` required.